test/components/dashboard/expand-students_spec.js

```diff
@@ -1,25 +1,25 @@
 import React from 'react'
 import { assert, expect } from 'chai'
 import { describe, it } from 'mocha'
-import { shallow } from 'enzyme'
+import { mount } from 'enzyme'
 import sinon from 'sinon'
 import { fromJS } from 'immutable'
 import ExpandStudents from '../../../js/components/dashboard/expand-students'
 
 describe('<ExpandStudents />', () => {
   it('should be labelled correctly on load', () => {
-    const wrapper = shallow(<ExpandStudents />)
-    assert(wrapper.find('Button').contains('Open Students'))
+    const wrapper = mount(<ExpandStudents />)
```

Comment:
The only downside of the new sub-component approach is that it means I need to full `mount` the parent in testing instead of a `shallow` mount.

Reply:
You could export `ExpandStudentsButton` too and test it separately using `shallow` (since it's unit testing). Perhaps it's never too many tests, although I think this is getting pretty far (testing specific labels).

You could also use `render`. I don't think you need full `mount` here.
Also, I'm not sure why there's division between `ExpandStudentsButton` and `ExpandStudents`. Not saying it's wrong, it's absolutely fine, but I think the reason to divide component would be when they get too big or e.g. to make testing easier. It's all good, just replying to your comment.

Reply:
Thanks @pjanik! I'll try it with `render` instead, I'm still trying to figure out all the differences...

About the new component - the idea was to have an `onClick` function I could call with different parameters without binding them in `props`. I made a comment about it above but it might have gotten lost, I was following this suggestion I found from `eslint`:

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#protips

Could be a useful pattern moving forward!

Reply:
FYI, they're mixing two very separate things, and take advantage of non-existing, old React API (`createReactClass`).

I agree with the first part - using `onClick={someFunction.bind(...)}` is bad. This is exactly the same as using arrow expresion  `onClick={() => { ... }}` which I keep constantly pointing out. And it's bad exactly due to the same reason. It creates a new function every time.

Adding a child component in this particular case also makes sense, as otherwise you would need to create some list of bound functions in the constructor - weird.

But then the only reason that they don't need to bind handler in child component is because `createReactClass` automatically binds all the methods. Note that you had to bind `onClick` in your `ExpandStudentsButton` class anyway. And you could have done the same thing in the parent component. I'd say there's no difference in this case.

But all in all, we're talking about really small things. ;) Althoug having some deeper understanding how React works is useful for sure.